mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassin.java

```diff
@@ -18,44 +18,43 @@
  ****************************************************************/
 package org.apache.james.mailbox.spamassassin;
 
-import java.io.InputStream;
-import java.util.List;
+import java.util.Optional;
 
 import javax.inject.Inject;
 
 import org.apache.james.core.Username;
 import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.spamassassin.MessageToLearn;
 import org.apache.james.spamassassin.SpamAssassinInvoker;
 import org.apache.james.util.Host;
 
-import com.github.fge.lambdas.Throwing;
+import reactor.core.publisher.Mono;
 
 public class SpamAssassin {
 
-    private final MetricFactory metricFactory;
-    private final SpamAssassinConfiguration spamAssassinConfiguration;
+    private final Optional<SpamAssassinInvoker> invoker;
 
     @Inject
     public SpamAssassin(MetricFactory metricFactory, SpamAssassinConfiguration spamAssassinConfiguration) {
```

Comment:
Already done in #3281

Reply:
Yes, we have already seen that. There will be some merge to be done when merging the second PR